### PR TITLE
[cmake] update cmake test port

### PR DESCRIPTION
**Pull request by Neumann-A**
required for transition to qt6

### CODE REVIEW

**Comment by NancyLi1013** [comment]
The failures on x64-linux:
```
c++: error: /mnt/vcpkg-ci/installed/x64-linux/debug/lib//libglib-2.0.a: No such file or directory
c++: error: /mnt/vcpkg-ci/installed/x64-linux/debug/lib//libglib-2.0.a: No such file or directory
c++: error: /mnt/vcpkg-ci/installed/x64-linux/debug/lib//libglib-2.0.a: No such file or directory
```

File: scripts/test_ports/cmake/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "cmake",
+  "version-string": "3.20.0",
+  "description": "CMake is an open-source, cross-platform family of tools designed to build, test and package software.",
+  "homepage": "https://cmake.org/",
+  "dependencies": [
+    "bzip2",
+    "curl",
+    "expat",
+    "expat",
```
Comment (Neumann-A) [inline scripts/test_ports/cmake/vcpkg.json:10]:
```suggestion
```

**Comment by Thieum** [comment]
See also #16845 where this test port seems to conflict with a newer version of libarchive.

**Comment by Neumann-A** [comment]
@Thieum have you reported it to upstream cmake?

**Comment by Thieum** [comment]
@Neumann-A no, I wasn't sure if it was a cmake issue, or a vcpkg issue. I didn't try to build cmake outside of the vcpkg context to tell if it can do so with the newest version of libarchive. I got mixed results with the automated tests: static builds seem to pass, but not the dynamic one.

**Pull request merged**